homeassistant/components/simplisafe/__init__.py

```diff
@@ -248,7 +248,7 @@ def _async_register_base_station(
 
     # Check for an old system ID format and remove it:
     if old_base_station := device_registry.async_get_device(
-        identifiers={(DOMAIN, system.system_id)}  # type: ignore[arg-type]
```

Comment:
Including this to satisfy `pre-commit`.

Reply:
I am interested in why this was here, because it would probably not have worked as we're creating devices with a str and querying them with a number

Reply:
That strict typing wasn't always required; guessing it's a very old relic.

Reply:
But I mean more like, is this code still relevant? Because this query would not fetch any new devices, so this didn't work for the last while

Reply:
I think we can remove this migration now, it was added 2 years ago

Reply:
Actually, I see what's going on: this typing is "newer" (compared to when this was originally implemented), and this logic was in place to ensure that a string was used.